cmd/main.go

```diff
@@ -1,15 +1,30 @@
 package main
 
 import (
+	"fmt"
 	"os"
 
 	"github.com/kyma-project/cli/cmd/kyma"
 	"github.com/kyma-project/cli/internal/cli"
+	"github.com/spf13/cobra/doc"
 )
 
+var AutoGenDoc string
+
+const DocsTargetDir = "./docs"
+
 func main() {
 	command := kyma.NewCmd(cli.NewOptions())
 
+	if AutoGenDoc == "true" {
```

Comment:
I find it quite strange to have the docs generation on the main command, we have to compile the whole CLI just to generate a couple of MD files.

It would be better to have a docs command or ideally use `go generate` so we do not need to compile and run things.

Reply:
I'm afraid this is the only way of doing that. https://github.com/spf13/cobra/blob/master/doc/md_docs.md

Reply:
hmm.. that is not what I understand from that document.

What I see is that you need to instantiate the command tree you want to document, but it is not required to do it in the exact same place right? We can have a doc generation specific main program which instantiates the Kyma root command and documents it.

Reply:
Moved generation to a separate binary `/cmd/gendocs/`

Reply:
looks great thanks!